scripts/scil_score_tractogram.py

```diff
@@ -0,0 +1,544 @@
+    p.add_argument('gt_bundles', nargs='+',
+                   help="Bundles ground truth(.trk, .nii or .nii.gz).")
+    p.add_argument('--gt_endpoints', nargs='+',
+                   help="Bundles endpoints, both bundle's ROIs\
+                       (.nii or .nii.gz).")
+    p.add_argument('--gt_tails', nargs='+',
+                   help="Bundles tails, bundle's first ROI(.nii or .nii.gz).")
+    p.add_argument('--gt_heads', nargs='+',
+                   help="Bundles heads, bundle's second ROI(.nii or .nii.gz).")
+
+    # p.add_argument('--dilate_endpoints', metavar='NB_PASS',
+    #                help='heuristic')
+
+    p.add_argument('--gt_config', metavar='FILE',
+                   help=".json dict to specify bundles streamlines min, \
+                    max length and max angles.")
+    p.add_argument('--out_dir', default='gt_out/',
+                   help="Output directory")
+
+    p.add_argument('--wrong_path_as_separate', action='store_true',
+                   help="Separates streamlines that go outside of the ground \
+                        truth mask from true connections, outputs as \
+                        *_wpc.trk.")
+
+    add_reference_arg(p)
+    add_overwrite_arg(p)
+    add_json_args(p)
+
+    return p
+
+
+def find_tc_pos(tc_filenames, filename):
+    if filename in tc_filenames:
+        return True, tc_filenames.index(filename)
+    return False, None
+
+
+def get_binary_maps(streamlines, dimensions, sft):
+    if not len(streamlines):
+        return np.zeros(dimensions), np.zeros(dimensions)
+    elif len(streamlines) == 1:
+        streamlines = [streamlines]
+    tmp_sft = StatefulTractogram.from_sft(streamlines, sft)
+    tmp_sft.to_vox()
+    tmp_sft.to_corner()
+    tmp_sft.remove_invalid_streamlines()
+
+    if len(tmp_sft) == 1:
+        return np.zeros(dimensions), np.zeros(dimensions)
+
+    bundles_voxels = compute_tract_counts_map(tmp_sft.streamlines,
+                                              dimensions).astype(np.int16)
+
+    endpoints_voxels = get_endpoints_density_map(tmp_sft.streamlines,
+                                                 dimensions).astype(np.int16)
+
+    bundles_voxels[bundles_voxels > 0] = 1
+    endpoints_voxels[endpoints_voxels > 0] = 1
+
+    return bundles_voxels, endpoints_voxels
+
+
+def identify_overlapping_roi(mask_1, mask_2, overlapping_roi):
+    if not os.path.isfile(mask_1):
+        raise ValueError('Input file {} does not exist.'.format(mask_1))
+    roi_1 = nib.load(mask_1)
+    roi1 = roi_1.get_fdata(dtype=np.float64)
+
+    if not os.path.isfile(mask_2):
+        raise ValueError('Input file {} does not exist.'.format(mask_2))
+    roi_2 = nib.load(mask_2)
+    roi2 = roi_2.get_fdata(dtype=np.float64)
+
+    rois = [roi1, roi2]
+    overlap = intersection(rois, roi_1)
+    nb_voxels = np.count_nonzero(overlap)
+
+    if nb_voxels > 0:
+        overlapping_roi.append((mask_1, mask_2))
+        overlapping_roi.append((mask_2, mask_1))
+
```

Comment:
return missing?

Reply:
I think the function uses side-effets instead of an explicit return, as `append` happens in-place.